cg/meta/demultiplex/demux_post_processing.py

```diff
@@ -73,129 +80,99 @@ def transfer_flow_cell(
 
         This function:
             - Parses and validates the flow cell directory data
             - Stores the flow cell in the status database
             - Stores sequencing metrics in the status database
             - Updates sample read counts in the status database
             - Stores the flow cell data in the housekeeper database
+            - Creates a delivery file in the flow cell directory
 
         Args:
             flow_cell_directory_name (str): The name of the flow cell directory to be finalized.
 
         Raises:
             FlowCellError: If the flow cell directory or the data it contains is not valid.
-
-        Note:
-            This method does not return anything but updates the status and housekeeper databases
-            with data from the flow cell directory.
         """
 
         LOG.info(f"Finish flow cell {flow_cell_directory_name}")
 
         flow_cell_directory_path: Path = Path(self.demux_api.out_dir, flow_cell_directory_name)
-        bcl_converter: str = self.get_bcl_converter_name(
-            flow_cell_directory=flow_cell_directory_path
-        )
+        bcl_converter: str = get_bcl_converter_name(flow_cell_directory_path)
 
-        parsed_flow_cell: FlowCellDirectoryData = self.parse_flow_cell_directory_data(
+        parsed_flow_cell: FlowCellDirectoryData = parse_flow_cell_directory_data(
             flow_cell_directory=flow_cell_directory_path,
             bcl_converter=bcl_converter,
         )
 
-        self.store_flow_cell_in_status_db(parsed_flow_cell=parsed_flow_cell)
-
-        self.add_flow_cell_data_to_housekeeper(
-            flow_cell_name=parsed_flow_cell.id,
-            flow_cell_directory=flow_cell_directory_path,
-        )
-
-        self.add_sample_lane_sequencing_metrics_for_flow_cell(
-            flow_cell_directory=flow_cell_directory_path, bcl_converter=bcl_converter
-        )
-
-        self.update_sample_read_counts(flow_cell_data=parsed_flow_cell)
+        try:
+            self.store_flow_cell_data(parsed_flow_cell)
+        except Exception as e:
+            LOG.error(f"Failed to store flow cell data: {str(e)}")
+            raise
 
-        self.create_delivery_file_in_flow_cell_directory(
-            flow_cell_directory=flow_cell_directory_path
-        )
+        create_delivery_file_in_flow_cell_directory(flow_cell_directory_path)
 
-    def create_delivery_file_in_flow_cell_directory(self, flow_cell_directory: Path) -> None:
-        Path(flow_cell_directory, DemultiplexingDirsAndFiles.DELIVERY).touch()
+    def store_flow_cell_data(self, parsed_flow_cell: FlowCellDirectoryData) -> None:
+        """Store data from the flow cell directory in status db and housekeeper."""
+        self.store_flow_cell_data_in_status_db(parsed_flow_cell)
+        self.store_sequencing_metrics_for_flow_cell(parsed_flow_cell)
+        self.update_sample_read_counts(parsed_flow_cell)
+        self.store_flow_cell_data_in_housekeeper(parsed_flow_cell)
```

Comment:
The logic contained in these methods should to a large extent be extracted to separate modules, such as 
```
FlowCellTransferApi
FlowCellProcessingApi
```
etc. but that will have to be done in separate PR:s.